### DISCUSSION TITLE
Questions regarding graphs with multiple edges

### DISCUSSION THREAD

**Discussion by yestinchen** [pyg-team/pytorch_geometric#4733] [Q&A] [answered]
Hi,

Thanks for devloping such a great platform for GNN.

I would like to use GNN to learn from a graph with multiple edges and I haven't found any discussions from the document regarding multiple edges between two nodes.
I am looking at the implementation of `GeneralConv` Layer, there are some questions I would like to confirm:
1. It seems that to support multiple edges, we only need to add duplicate entries in `edge_index`, and add corresponding edge ettributes in the `edge_attr`?
2. Can we do the same thing for graphs with self-loops? Is it okay if say I have one node with multiple self-loop edges but with different features? Assuming all edges are of the same type. 

Thank you!

**Comment by rusty1s** [comment, marked as answer]
Yes, your understanding is absolutely correct.

**Reply by yestinchen** [reply]
Thanks for the quick response!
